source/Halibut/Transport/Observability/ByteCountingStream.cs

```diff
@@ -45,6 +46,14 @@ protected override void Dispose(bool disposing)
             }
         }
 
+        public override async ValueTask DisposeAsync()
```

Comment:
The Dispose and DisposeAsync differ in their implementation a bit. Should they be consistent?